web/src/routes/UsageLogs/UsageLogsGraph.tsx

```diff
@@ -76,48 +80,63 @@ export default function UsageLogsGraph(props: UsageLogsGraphProps) {
     return legends;
   }
 
-  return (
-    <Flex grow={{default: 'grow'}}>
-      <FlexItem>
-        <Chart
-          key={props.starttime + props.endtime}
-          containerComponent={
-            <ChartVoronoiContainer
-              labels={({datum}) => `${datum.name}: ${datum.y}`}
-              constrainToVisibleArea
-            />
-          }
-          domain={{
-            x: [new Date(props.starttime), new Date(props.endtime)],
-            y: [0, maxRange],
-          }}
-          legendOrientation={
-            getLegendData().length >= 12 ? 'horizontal' : 'vertical'
-          }
-          legendPosition={getLegendData().length >= 12 ? 'bottom' : 'right'}
-          legendData={getLegendData()}
-          legendAllowWrap
-          name="usage-logs-graph"
-          padding={{
-            bottom: 10 * getLegendData().length,
-            left: 80,
-            right: 500, // Adjusted to accommodate legend
-            top: 50,
-          }}
-          domainPadding={{x: 5 * Object.keys(logData).length}}
-          height={400}
-          width={1250}
-          scale={{x: 'time', y: 'linear'}}
-        >
-          <ChartAxis fixLabelOverlap />
-          <ChartAxis dependentAxis showGrid />
-          <ChartGroup offset={11}>
-            {Object.keys(logData).map((logKind, index) => (
-              <ChartBar data={logData[logKind]} key={index} />
-            ))}
-          </ChartGroup>
-        </Chart>
-      </FlexItem>
-    </Flex>
-  );
+  if (getLegendData().length === 0) {
+    return (
+      <Flex grow={{default: 'grow'}} style={{margin: '50px'}}>
+        <FlexItem>
+          <p>No data to display.</p>
+        </FlexItem>
+      </Flex>
+    );
+  } else
+    return (
+      <Flex grow={{default: 'grow'}}>
+        <FlexItem>
+          <Chart
+            key={props.starttime + props.endtime}
+            containerComponent={
+              <ChartVoronoiContainer
+                labels={({datum}) => `${datum.name}: ${datum.y}`}
+                constrainToVisibleArea
+              />
+            }
+            domain={{
+              x: [new Date(props.starttime), new Date(props.endtime)],
+              y: [0, maxRange],
+            }}
+            legendAllowWrap
+            legendComponent={
+              <ChartLegend
+                data={getLegendData()}
+                itemsPerRow={8}
+                theme={useTheme()}
+              />
+            }
+            legendPosition="right"
+            legendOrientation={
+              getLegendData().length >= 12 ? 'horizontal' : 'vertical'
+            }
+            name="usage-logs-graph"
+            padding={{
+              bottom: 20 * getLegendData().length,
+              left: 80,
+              right: 500, // Adjusted to accommodate legend
+              top: 50,
+            }}
+            domainPadding={{x: 5 * Object.keys(logData).length}}
+            height={400}
```

Comment:
Ther x-axis was still covered up for me, I had to change this to 450 to make it visible